Pod/Classes/WPMediaCollectionViewController.m

```diff
@@ -823,4 +827,30 @@ - (void)assetViewController:(WPAssetViewController *)assetPreviewVC selectionCha
     }
 }
 
+- (void)assetViewController:(WPAssetViewController *)assetPreviewVC failedWithError:(NSError *)error
+{
+    BOOL needToPop = YES;
+    if (self.navigationController.topViewController == self) {
+        needToPop = NO;
+    }
+    NSString *errorDetails = error.localizedDescription;
+    NSError *underlyingError = error.userInfo[NSUnderlyingErrorKey];
+    if (underlyingError) {
+        errorDetails = underlyingError.localizedDescription;
+    }
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Media preview failed.", @"Alert title when there is issues loading an asset to preview.")
+                                                                             message:errorDetails
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction *dismissAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Dismiss", @"Action to show on alert when view asset fails.") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        if (needToPop) {
+            [self.navigationController popViewControllerAnimated:YES];
+        } else {
+            [self dismissViewControllerAnimated:YES completion:nil];
+        }
+    }];
+    [alertController addAction:dismissAction];
+
+    [self presentViewController:alertController animated:YES completion:nil];
```

Comment:
So this isn't working right with 3D Touch. I'm getting no alert appear, and 

`Warning: Attempt to present <UIAlertController: 0x1004c3700>  on <WPMediaCollectionViewController: 0x10031dff0> which is already presenting (null)`

in the console. I think because the peek preview is still visible and acting as a presented view controller. Perhaps we should dismiss it first, and then present the alert.